src/main/scala/uk/gov/nationalarchives/pdi/test/QueryManager.scala

```diff
@@ -28,12 +28,17 @@ import uk.gov.nationalarchives.pdi.test.helpers.IOHelper.findFile
 
 import java.nio.file.Path
 import scala.annotation.tailrec
+import scala.collection.mutable
 import scala.util.{ Failure, Success, Try, Using }
+import org.apache.jena.query.QuerySolution
+import scala.jdk.CollectionConverters._
 
 /** Used to execute SPARQL queries against RDF files output by Pentaho Kettle during transformation testing
   */
 object QueryManager {
 
+  case class RecordSet(results: List[Map[String, Object]])
```

Comment:
It might be better to call the List `contents` as the RecordSet itself could be described as a result.